# Part 1.

## (a)

In [14]:
from sklearn.datasets import load_digits
import matplotlib.pyplot as plt
import numpy as np

load_digits_dataset = load_digits()
print(load_digits_dataset)


{'data': array([[ 0.,  0.,  5., ...,  0.,  0.,  0.],
       [ 0.,  0.,  0., ..., 10.,  0.,  0.],
       [ 0.,  0.,  0., ..., 16.,  9.,  0.],
       ...,
       [ 0.,  0.,  1., ...,  6.,  0.,  0.],
       [ 0.,  0.,  2., ..., 12.,  0.,  0.],
       [ 0.,  0., 10., ..., 12.,  1.,  0.]], shape=(1797, 64)), 'target': array([0, 1, 2, ..., 8, 9, 8], shape=(1797,)), 'frame': None, 'feature_names': ['pixel_0_0', 'pixel_0_1', 'pixel_0_2', 'pixel_0_3', 'pixel_0_4', 'pixel_0_5', 'pixel_0_6', 'pixel_0_7', 'pixel_1_0', 'pixel_1_1', 'pixel_1_2', 'pixel_1_3', 'pixel_1_4', 'pixel_1_5', 'pixel_1_6', 'pixel_1_7', 'pixel_2_0', 'pixel_2_1', 'pixel_2_2', 'pixel_2_3', 'pixel_2_4', 'pixel_2_5', 'pixel_2_6', 'pixel_2_7', 'pixel_3_0', 'pixel_3_1', 'pixel_3_2', 'pixel_3_3', 'pixel_3_4', 'pixel_3_5', 'pixel_3_6', 'pixel_3_7', 'pixel_4_0', 'pixel_4_1', 'pixel_4_2', 'pixel_4_3', 'pixel_4_4', 'pixel_4_5', 'pixel_4_6', 'pixel_4_7', 'pixel_5_0', 'pixel_5_1', 'pixel_5_2', 'pixel_5_3', 'pixel_5_4', 'pixel_5_5', 'pixel_

## (b)

The "load_digits" dataset contains a collection of 8x8 drawn images of digits ranging from 0-10. The images are kept as matrix arrays, containing values ranging from 0.0-16.0. It has 1797 instances in total. For each drawn image(instance) there are 64 pixels (features).

In [22]:
print("No. of classes:", len(np.unique(load_digits_dataset.target)))
print("Classes:", np.unique(load_digits_dataset.target))
print("Pixel min value:", load_digits_dataset.data.min())
print("Pixel max value:", load_digits_dataset.data.max())
print("No. of pixels for each image (features):", load_digits_dataset.data.shape[1])

No. of classes: 10
Classes: [0 1 2 3 4 5 6 7 8 9]
Pixel min value: 0.0
Pixel max value: 16.0
No. of pixels for each image (features): 64


## (c)

LALLALALALAL